lib/ash_json_api/json_schema/json_schema.ex

```diff
@@ -316,48 +322,53 @@ defmodule AshJsonApi.JsonSchema do
     }
   end
 
-  defp resource_attribute_type(%{type: Ash.Type.String}) do
+  @doc false
+  def resource_attribute_type(%{type: Ash.Type.String}) do
     %{
       "type" => "string"
     }
   end
 
-  defp resource_attribute_type(%{type: Ash.Type.Boolean}) do
+  def resource_attribute_type(%{type: Ash.Type.Boolean}) do
     %{
-      "type" => ["boolean", "string"],
-      "match" => "^(true|false)$"
+      "oneOf" => [
```

Comment:
In swagger 3.0, the `type` field isn't allowed to be an array. Fixed in 3.1, but I've targeted 3.0 in this PR as 3.1 doesn't seem to be universally supported by tooling (notably SwaggerUI).